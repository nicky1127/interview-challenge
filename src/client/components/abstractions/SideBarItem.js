import React from "react";
import PropTypes from "prop-types";

const SideBarItem = props => {
  const { item } = props;

  let content;

  if (Array.isArray(item.dietaries) && item.dietaries.length > 0) {
    content = item.dietaries.map((dietary, index) => (
      <span key={`sidebarItem_dietary-${index}`} className="dietary" data-testid='sideBarItem'>
        {dietary}
      </span>
    ));
  }
  return (
    <li className="item">
      <h2>{item.name}</h2>
      <p>{content}</p>
    </li>
  );
};

SideBarItem.defaultProps = { item: {}};

SideBarItem.propTypes = {
    item: PropTypes.exact({
        name: PropTypes.string.isRequired,
        id: PropTypes.number.isRequired,
        dietaries:PropTypes.arrayOf(PropTypes.string).isRequired
      }).isRequired
};

export default SideBarItem;
