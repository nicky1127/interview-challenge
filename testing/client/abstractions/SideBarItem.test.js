import React from "react";
import { render, cleanup } from "@testing-library/react";
import SideBarItem from "client/components/abstractions/SideBarItem";

beforeEach(() => {
  jest.clearAllMocks();
  cleanup();
});

const props = { item: { name: "test", id: 1, dietaries: ["test"] } };

jest.useFakeTimers();

describe("SideBarItem", () => {
  it("Snapshot matchs", () => {
    const tree = renderer.create(<SideBarItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("SideBarItem should render one [data-testid=sideBarItem]", () => {
    const { getByTestId } = render(<SideBarItem {...props} />);
    const container = getByTestId("sideBarItem");
    expect(container).toBeInTheDocument();
  });
});
