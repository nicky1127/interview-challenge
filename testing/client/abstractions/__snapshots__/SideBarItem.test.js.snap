// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarItem Snapshot matchs 1`] = `
<li
  className="item"
>
  <h2>
    test
  </h2>
  <p>
    <span
      className="dietary"
      data-testid="sideBarItem"
    >
      test
    </span>
  </p>
</li>
`;
